Add unit-test-in-progress for suffix tree.

learner-long-life opened the pull request:
Add unit tests for suffix tree in a dedicated `tests` directory tree, with test submodules that mirror the code submodules, at least for `edit_distance` and `suffix_tree` so far.

learner-long-life committed:
Add unit-test-in-progress for suffix tree.

AbyssalRemark commented on src/data_structures/suffix_trees.rs:
```diff
@@ -7,7 +7,7 @@ const ROOT: usize = 1;
 enum End {
     Root,
-    End,
+    EOS,
```
Good name change.  I would have perhapsed changed enum End to enum End_state. but that's just me.

learner-long-life replied:
@Gavin-Bowers and @tw-edu will be making more changes concurrently touching this enum, so it will be a separate task to convince them from this PR.

AbyssalRemark commented on src/lib.rs:
```diff
@@ -3,7 +3,7 @@ pub mod data_structures;
 
 #[cfg(test)]
-mod tests {
+mod edit_distance_tests {
```
If this works then more descriptive is better. BUT, this should probably be in the file for what its testing.

learner-long-life replied:
It does work, and is necessary for multiple sub-module "integration" tests that appear to be how Rust / Cargo supports them currently.

https://stackoverflow.com/a/37626704
https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests

AbyssalRemark reviewed (changes requested):
I think moving this test code to suffix tree file makes a lot more sense. There is the possibility to tests in there own test directory. But as we haven't worked that out. putting it in suffix trees makes the most sense.

learner-long-life committed:
Move tests into their own directory.

Create sub modules for cargo test to find the new tests. Add a simpler one-node suffix tree test. Add default binary.

learner-long-life commented:
@AbyssalRemark please take another look when you can.
